reframe/utility/sanity.py

```diff
@@ -610,11 +616,26 @@ def extractall(patt, filename, tag=0, conv=None, encoding='utf-8'):
         Group ``0`` refers always to the whole match.
         Since the file is processed line by line, this means that group ``0``
         returns the whole line that was matched.
-    :arg conv: A callable that takes a single argument and returns a new value.
-        If provided, it will be used to convert the extracted values before
-        returning them.
-    :returns: A list of the extracted values from the matched regex.
+    :arg conv: A callable or iterable of callables taking a single argument
+        and returning a new value.
+        If provided, and is not an iterable it will be used to convert
+        the extracted values for all the capturing groups of ``tag``
+        returning the converted values.
+        If an iterable of callables is provided, each one will be used to
+        convert the corresponding extracted capturing group of `tag`.
+        If more callables functions than the corresponding capturing groups of
+        ``tag`` are provided, the last conversion function is used for the
+        remaining capturing groups.
+    :returns: A list of the extracted values from the matched regex if ``tag``
+        converted using the ``conv`` callable if ``tag`` is a single value.
+        In case of multiple capturing groups, a list of tuples where each one
+        contains the extracted capturing converted using the corresponding
+        callable of ``conv``.
```

Comment:
```suggestion
    :returns: A list of tuples of converted values extracted from the capturing groups specified in ``tag``, if ``tag`` is an iterable. Otherwise, a list of the converted values extracted from the single capturing group specified in ``tag``.
```